src/Aspire.Hosting/ApplicationModel/ResourceExtensions.cs

```diff
@@ -65,6 +65,88 @@ public static bool TryGetEnvironmentVariables(this IResource resource, [NotNullW
         return TryGetAnnotationsOfType(resource, out environmentVariables);
     }
 
+    /// <summary>
+    /// Get the environment variables from the given resource.
+    /// </summary>
+    /// <remarks>
+    /// This method is useful when you want to make sure the environment variables are added properly to resources, mostly in test situations.
+    /// This method has asynchronous behavior when <paramref name = "applicationOperation" /> is <see cref="DistributedApplicationOperation.Run"/>
+    /// and environment variables were provided from <see cref="IValueProvider"/> otherwise it will be synchronous.
+    /// </remarks>
+    /// <param name="resource">The resource to get the environment variables from.</param>
+    /// <param name="applicationOperation">The context in which the AppHost is being executed.</param>
+    /// <returns>The environment variables retrieved from the resource.</returns>
+    /// <example>
+    /// Using <see cref="GetEnvironmentVariableValuesAsync(IResourceWithEnvironment, DistributedApplicationOperation)"/> inside
+    /// a unit test to validate environment variable values.
+    /// <code>
+    /// var builder = DistributedApplication.CreateBuilder();
+    /// var container = builder.AddContainer("elasticsearch", "library/elasticsearch", "8.14.0")
+    ///  .WithEnvironment("discovery.type", "single-node")
+    ///  .WithEnvironment("xpack.security.enabled", "true")
+    ///  .WithEnvironment(context =>
+    ///  {
+    ///      context.EnvironmentVariables["ELASTIC_PASSWORD"] = "123456";
```

Comment:
Maybe this so credscan does not complain:
```suggestion
    ///      context.EnvironmentVariables["ELASTIC_PASSWORD"] = "p@ssw0rd1";
```

Or we can add this to `./.config/CredScanSuppressions.json`.